### ISSUE TITLE
Work directly with the log-pdf

### ISSUE THREAD

**Issue by mberaha** [mauriciogtec/AdaptiveRejectionSampling.jl#10]
Is it possible to pass directly the log-density instead of the density itself? Due to some numerical instability, I sometimes get some NaNs or Infs in the weights.

**Comment by mauriciogtec** [comment]
I've noticed this issue before.
Can you post an example where NaNs are generated to use it as a test?
I can work on a PR.

**Comment by mberaha** [comment]
Sure, this is a rather challenging distribution that I'm working with

```
n = 50
k = 10
alpha = 0.5
tau = 0.5
theta = 1.0


logf(v) = n * v - (n - k * alpha) * logsumexp([v, log(tau)]) -
    theta / alpha * ( (tau + exp(v) )^alpha )


f(u) = exp(logf(u))


support = (-Inf, Inf)
sampler = RejectionSampler(f, support, search_range=(-10.0, 10.0))
sim = run_sampler!(sampler, 10000) 
```

Note that I'm already defining `logf` and then passing `exp(logf)` since defining directly the density leads to much greater numerical instability


If I plug-in `n=600, k=100` instead I get the error

```
AssertionError: cutpoints must be ordered
```

**Comment by mauriciogtec** [comment]
@mberaha  This issue should be fixed with PR #11 and included in v0.2.0, which should be available in the Julia registry soon (I've triggered the release), but I don't know how much time it will take to be in the package registry.

I have added your example in the README. There will likely be some annoying repeated warnings. I created an issue for this to be included in the next release.

Feel free to raise more issues as you encounter them in your work.
